apps/e2e/cypress/e2e/talentsearch/submit-applicationRevamp.cy.js

```diff
@@ -0,0 +1,369 @@
+                  publishingGroup: PublishingGroup.ItJobs,
+                  screeningQuestions: {
+                    create: [
+                      {
+                        question: { en: "Question EN", fr: "Question FR" },
+                        sortOrder: 1,
+                      },
+                    ],
+                  },
+                });
+                cy.publishPool(testPoolId);
+              });
+            });
+        });
+      });
+
+    cy.get("@testUser").then((user) => {
+      cy.loginBySubject(user.sub);
+    });
+    cy.visit("/en/browse/pools");
+
+    // Browse pools page - placeholder so it could change
+    cy.wait("@gqlbrowsePoolsQuery");
+    cy.findByRole("heading", { name: /browse jobs/i })
+      .should("exist")
+      .and("be.visible");
+    cy.get("@testPoolId").then((testPoolId) => {
+      cy.get(`a[href*="${testPoolId}"]`).click(); // names could be duplicated - choose the one with the right ID in the url
+    });
+
+    // Pool poster page
+    cy.wait("@gqlgetPoolQuery");
+    cy.findByRole("heading", { name: /Apply now/i })
+      .should("exist")
+      .and("be.visible");
+    cy.findAllByRole("link", { name: /Apply for this process/i })
+      .first()
+      .click();
+    cy.wait("@gqlcreateApplicationMutation");
+
+    // Welcome page - step one
+    cy.wait("@gqlGetApplicationQuery");
+    cy.findByRole("heading", { name: /Welcome, Cypress/i })
+      .should("exist")
+      .and("be.visible");
+    cy.findByRole("button", { name: /Let's go!/i }).click();
+    cy.wait("@gqlUpdateApplicationMutation");
+
+    // Review profile page - step two
+    cy.wait("@gqlGetApplicationQuery");
+    cy.findByRole("heading", { name: /Review your profile/i })
+      .should("exist")
+      .and("be.visible");
+    cy.contains(/Just a heads up!/i).should("not.exist"); // error summary message on profile page not present
+    cy.findByRole("button", { name: /Save and continue/i }).click();
+    cy.wait("@gqlUpdateApplicationMutation");
+
+    // Review resume page - step three
+    cy.wait("@gqlGetApplicationQuery");
+    cy.findByRole("heading", { name: /Great work! On to your résumé./i })
+      .should("exist")
+      .and("be.visible");
+
+    // Attempt skipping to review
+    cy.url().then((url) => {
+      const reviewResumeIntroUrl = url;
+      cy.expect(reviewResumeIntroUrl).to.have.string("resume/introduction");
+      const hackedUrl = reviewResumeIntroUrl.replace(
+        "resume/introduction",
+        "review",
+      );
+      cy.visit(hackedUrl);
+    });
+    cy.findByRole("heading", {
+      name: /Uh oh, it looks like you jumped ahead!/i, // message for skipping steps
+    })
+      .should("exist")
+      .and("be.visible");
+    cy.findByRole("link", {
+      name: /Return to the last step I was working on/i,
+    }).click();
+    cy.findByRole("heading", { name: /Create your résumé/i }) // returned to resume step
+      .should("exist")
+      .and("be.visible");
```

Comment:
Could this be a separate test case? Something like `it("prevents users from skipping steps")`

Reply:
Yeah... if you aren't sure it'd work might be best to experiment with it first. It wouldn't effect things that much I imagine it'll be low priority